robocorp-code/vscode-client/src/pythonExtIntegration.ts

```diff
@@ -45,24 +62,54 @@ export async function autoUpdateInterpreter(docUri: Uri) {
     }
 }
 
-export async function installPythonInterpreterCheck(context: ExtensionContext) {
-    context.subscriptions.push(
-        window.onDidChangeActiveTextEditor(async (event: TextEditor) => {
-            try {
-                // Whenever the active editor changes we update the Python interpreter used (if needed).
-                let docUri = event?.document?.uri;
-                if (docUri) {
-                    await autoUpdateInterpreter(docUri);
+export async function installWorkspaceWatcher(context: ExtensionContext) {
+    // listen for document changes
+    const checkIfFilesHaveChanged = workspace.onDidChangeTextDocument(async (event) => {
+        let docURI = event.document.uri;
+        if (event.document.isDirty && (isEnvironmentFile(docURI.fsPath) || isPythonFile(docURI.fsPath))) {
+            dirtyWorkspaceFiles.add(docURI.fsPath);
+        }
+    });
```

Comment:
The old auto-update on changes is still needed.

The use-case for it is having multiple tasks where each task has a different environment (so, by changing the open file the "current" interpreter must be changed).

Reply:
Got it. 
Will update.